### PR TITLE
New working machines

### PR DESCRIPTION
**Pull request by travistyoj**
Power Up Baseball (prototype) [Frank Cifaldi, Rich Whitehouse, Ryan
Holtz, Travis Brown, Dan Filner]

* itech32: Updated NVRAM mapping on '020 machines to cover full 128k.
  [Travis Brown]

### CODE REVIEW

File: src/mame/drivers/itech32.cpp [hunk 1]
```diff
@@ -1011,8 +1012,7 @@ void itech32_state::itech020_map(address_map &map)
 	map(0x500000, 0x5000ff).rw(FUNC(itech32_state::bloodstm_video_r), FUNC(itech32_state::bloodstm_video_w));
 	map(0x578000, 0x57ffff).nopr();             /* touched by protection */
 	map(0x580000, 0x59ffff).ram().w(m_palette, FUNC(palette_device::write32)).share("palette");
-	map(0x600000, 0x603fff).ram().share("nvram32");
-/* ? */ map(0x61ff00, 0x61ffff).nopw();            /* Unknown Writes */
+	map(0x600000, 0x61ffff).ram().share("nvram32");
 	map(0x680002, 0x680002).r(FUNC(itech32_state::itech020_prot_result_r));
```
Comment (aaronsgiles) [inline src/mame/drivers/itech32.cpp:1016]: Do we have proof that ALL itech020 boards had 128k of NVRAM? If not, then this change should only apply to pubball, which already had its own map.

File: src/mame/drivers/itech32.cpp [hunk 2]
```diff
@@ -4920,6 +5009,18 @@ Label1  bne.s       Label1          ; Infinite loop if result isn't 0x80
...
 
+void itech32_state::init_pubball()
+{
+	init_program_rom();
+	m_vram_height = 1024;
+	m_planes = 2;
+
+	m_maincpu->space(AS_PROGRAM).install_write_handler(0x300000, 0x300003, write8smo_delegate(*this, FUNC(itech32_state::color_w<0>)), 0x000000ff);
+	m_maincpu->space(AS_PROGRAM).install_write_handler(0x380000, 0x380003, write8smo_delegate(*this, FUNC(itech32_state::color_w<1>)), 0x000000ff);
+	m_maincpu->space(AS_PROGRAM).install_read_handler(0x180800, 0x180803, read32smo_delegate(*this, FUNC(itech32_state::trackball32_4bit_p1_r)));
```
Comment (aaronsgiles) [inline src/mame/drivers/itech32.cpp:5020]: The color_w installations here seem redundant with the pubball memory map above? And if it has its own memory map, might as well chuck the input customizations there as well.

File: src/mame/drivers/itech32.cpp (hunk 1 above)
Comment (MooglyGuy) [reply src/mame/drivers/itech32.cpp:1016]: All? No. Do we have proof WHICH itech020 games had 128k of NVRAM? Also no.

But the fact that the back end of the 128k range had enough writes to merit a nopw() handler would indicate that it was common enough, at least.

I specifically advised that Travis make this change, and I stand by it.

File: src/mame/drivers/itech32.cpp (hunk 2 above)
Comment (MooglyGuy) [reply src/mame/drivers/itech32.cpp:5020]: They are, I wasn't aware that these changes were being made. I will rectify the redundancy sometime this weekend when I have time. For now, it's more important that we have support for the game in mainline for both visibility and consistency with the media push that is about to begin. Work with me, here.

**Pull request merged**